server/src/main/java/org/opensearch/snapshots/RemoteStoreRestoreService.java

```diff
@@ -0,0 +1,278 @@
+import org.opensearch.common.UUIDs;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.indices.ShardLimitValidator;
+import org.opensearch.indices.replication.common.ReplicationType;
+import org.opensearch.repositories.IndexId;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_INDEX_UUID;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_VERSION_CREATED;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_REPLICAS;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_NUMBER_OF_SHARDS;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_SEGMENT_STORE_REPOSITORY;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REPLICATION_TYPE;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_STORE_ENABLED;
+
+/**
+ * Service responsible for restoring indices from remote
+ *
+ * @opensearch.internal
+ */
+public class RemoteStoreRestoreService implements ClusterStateApplier {
+
+    private static final Logger logger = LogManager.getLogger(RemoteStoreRestoreService.class);
+
+    private final ClusterService clusterService;
+
+    private final AllocationService allocationService;
+
+    private final MetadataCreateIndexService createIndexService;
+
+    private final MetadataIndexUpgradeService metadataIndexUpgradeService;
+
+    private final ShardLimitValidator shardLimitValidator;
+
+    private final ClusterSettings clusterSettings;
+
+    public RemoteStoreRestoreService(
+        ClusterService clusterService,
+        AllocationService allocationService,
+        MetadataCreateIndexService createIndexService,
+        MetadataIndexUpgradeService metadataIndexUpgradeService,
+        ClusterSettings clusterSettings,
+        ShardLimitValidator shardLimitValidator
+    ) {
+        this.clusterService = clusterService;
+        this.allocationService = allocationService;
+        this.createIndexService = createIndexService;
+        this.metadataIndexUpgradeService = metadataIndexUpgradeService;
+        if (DiscoveryNode.isClusterManagerNode(clusterService.getSettings())) {
+            clusterService.addStateApplier(this);
+        }
+        this.clusterSettings = clusterService.getClusterSettings();
+        this.shardLimitValidator = shardLimitValidator;
+    }
+
+    @Override
+    public void applyClusterState(ClusterChangedEvent event) {
+
+    }
```

Comment:
Why cluster state applier when this is no-op

Reply:
Copied the from RestoreService. Removed now, we dont need this for remote restore